src/test/java/seedu/address/model/student/ContactNumberTest.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.student;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class ContactNumberTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new ContactNumber(null));
+    }
+
+    @Test
+    public void constructor_invalidPhone_throwsIllegalArgumentException() {
+        String invalidPhone = "";
+        assertThrows(IllegalArgumentException.class, () -> new ContactNumber(invalidPhone));
+    }
+
+    @Test
+    public void isValidPhone() {
+        // null phone number
+        assertThrows(NullPointerException.class, () -> ContactNumber.isValidContact(null));
+
+        // invalid phone numbers
+        assertFalse(ContactNumber.isValidContact("")); // empty string
+        assertFalse(ContactNumber.isValidContact(" ")); // spaces only
+        assertFalse(ContactNumber.isValidContact("91")); // less than 3 numbers
+        assertFalse(ContactNumber.isValidContact("phone")); // non-numeric
+        assertFalse(ContactNumber.isValidContact("9011p041")); // alphabets within digits
+        assertFalse(ContactNumber.isValidContact("9312 1534")); // spaces within digits
```

Comment:
Might be abit too strict?

Reply:
Yea it probably is. However, ContactNumber is currently not being used so we could probably make regex changes if we decide to use it in our Student model in the future.

Reply:
I can just comment it out for now and we can decide on this when we decide to use the contactNumber.